taipy/gui/server.py

```diff
@@ -267,17 +267,25 @@ def _apply_patch(self):
             if not patcher.is_monkey_patched("time"):
                 monkey_patch(time=True)
 
-    def _get_random_port(self):  # pragma: no cover
+    def _get_random_port(self, port_auto_ranges: t.Optional[t.List[t.Union[int, t.Tuple[int, int]]]] = None):  # pragma: no cover
+        port_auto_ranges = port_auto_ranges or [(49152, 65535)]
+        random_weights = [
+            1 if isinstance(r, int) else r[1] - r[0] + 1 for r in port_auto_ranges
```

Comment:
Are you supposing that the first element of the tuple is lesser than the second one ?

Reply:
you are right on this one I guess, never know what the user might enter. I will get the abs value